### PR TITLE
[feature] Add CommitID and GeneratedBy in the Metadata Properties

### PR DESCRIPTION
**Pull request by stisac**
*Issue #, if available:* https://sim.amazon.com/issues/AML-114468

*Description of changes:*

1. Added CommitId and GeneratedBy in the Metadata Properties dict()
2. Added tests for the same in pipeline.py

*Testing done:*
tox -r -e py39 -- ~/github/sagemaker-python-sdk/tests/unit/sagemaker/workflow/test_pipeline.py 

Results:

collected 10 items                                                                                                                                                                                                                        

tests/unit/sagemaker/workflow/test_pipeline.py ..........

### CODE REVIEW

File: src/sagemaker/workflow/pipeline.py [hunk 1]
```diff
@@ -59,9 +62,15 @@ class Pipeline(Entity):
     parameters: Sequence[Parameter] = attr.ib(factory=list)
     steps: Sequence[Union[Step, StepCollection]] = attr.ib(factory=list)
     sagemaker_session: Session = attr.ib(factory=Session)
+    repo = Repo("~/.git")
```
Comment (staubhp) [inline src/sagemaker/workflow/pipeline.py:65]: You can't assume the SDK is in a git repo. This will throw `git.exc.NoSuchPathError` if it isn't.

Comment (staubhp) [reply src/sagemaker/workflow/pipeline.py:65]: Also this is reading from the user's home directory, not the directory root

**Review by staubhp — COMMENTED** [review]
You should add an integ test for this change in [https://github.com/aws/sagemaker-python-sdk/blob/master/tests/integ/test_workflow.py](test_workflow.py).

For faster iteration on testing you can compile the SDK (`python setup.py sdist`) and install it (`!pip install -U path/to/sagemaker.tar.gz`) for local testing

**Comment by stisac** [comment]
> You should add an integ test for this change in [https://github.com/aws/sagemaker-python-sdk/blob/master/tests/integ/test_workflow.py](test_workflow.py).
> 
> For faster iteration on testing you can compile the SDK (`python setup.py sdist`) and install it (`!pip install -U path/to/sagemaker.tar.gz`) for local testing

Sure

**Comment by shreyapandit** [comment]
Discussed with @stisac, their team plans to pick this up in near future

**Comment by shreyapandit** [comment]
Thanks for your contribution! Will be closing this PR because of inactivity and no plans to address open feedback in near future. Please discuss with team and re-open when it's ready to be taken forward.

**Pull request closed without merging**